lib/validation/schema.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Litmus.Validation.Schema do
+  @moduledoc false
+
+  @doc """
+  Module for checking and routing to individual schema specific validation
+  """
+
+  alias Litmus.Type.{Any, Boolean, Number, String}
+
+  @spec check_schema_error(map, map) :: tuple
+  def check_schema_error(data, schema) do
+    Enum.reduce_while(schema, %{}, fn {k, v}, acc ->
+      data
```

Comment:
I believe in your design doc you traversed the schema not the data so that errors would appear in the order based on the schema. Although it's a map, so order of iteration over keys isn't guaranteed anyway... Hmm. Any pros/cons to iterating over `data` vs `schema`?